pytest_inmanta/module/init.cf

```diff
@@ -21,19 +21,44 @@ entity Resource extends std::PurgeableResource:
         A test resource for unit and integration testing purposes.
 
         :param skip: When set to true the handler will raise a skipresource
-        :param fail: When set to true the handler will raise an exception to mark a failure
+        :param fail: When set to true the handler will raise an exception to mark a failure whenever it performs a CRUD
+            operation.
+        :param fail_deploy: When set to true the handler will raise an exception to mark a failure when it actually performs
+            a deploy operation, i.e. create, update, delete.
     """
     string name
     string desired_value
     bool skip=false
     bool fail=false
+    bool fail_deploy=false
     string agent="internal"
 end
 
 index Resource(name)
 
 implement Resource using std::none
 
+entity WrongDiffResource extends std::PurgeableResource:
```

Comment:
Any reason you make this an entirely new resource rather than an option in the other one?

Reply:
Oh that makes a lot more sense, I wanted to use a different handler so I had to make a different resource. But I can just make an option on the other one and overwrite the `calculate_diff` there

Reply:
Yes, I think that would be simpler. And in the body of `calculate_diff` just call `super().calculate_diff()` if the option is disabled.